### PR TITLE
Capped Mixing Length Turbulence

### PR DESCRIPTION
**Pull request by freiler**

### CODE REVIEW

**Comment by GiudGiud** [comment]
Thanks for the contribution, I ll review this soon!
Looked at the tests:
- rename INASDMixingLengthTurbulentViscosityMaterial.md to MixingLengthTurbulentViscosityMaterial.md
- new tests need to be moved from moose/test to moose/modules/navier_stokes/test

File: modules/navier_stokes/src/auxkernels/WallDistanceMixingLengthAux.C [hunk 1]
```diff
@@ -19,13 +19,17 @@ WallDistanceMixingLengthAux::validParams()
                              "proportional to the distance from the nearest wall.");
   params.addParam<std::vector<BoundaryName>>("walls", "Boundaries that correspond to solid walls");
   params.addParam<Real>("von_karman_const", 0.4, "");
+  params.addParam<Real>("von_karman_const_0", 0.09, "");
+  params.addParam<Real>("delta", 1e9, "");
```
Comment (GiudGiud) [inline modules/navier_stokes/src/auxkernels/WallDistanceMixingLengthAux.C:23]: all 3 need a description

File: modules/navier_stokes/src/auxkernels/WallDistanceMixingLengthAux.C [hunk 2]
```diff
@@ -62,6 +66,13 @@ WallDistanceMixingLengthAux::computeValue()
...
-  return _von_karman_const * std::sqrt(min_sq_dist);
+
+  if (std::sqrt(min_sq_dist) / _delta <= _von_karman_const_0 / _von_karman_const)
+  {
+    return _von_karman_const * std::sqrt(min_sq_dist);
+  }
+  else
+  {
+    return _von_karman_const_0 * _delta;
+  }
```
Comment (GiudGiud) [inline modules/navier_stokes/src/auxkernels/WallDistanceMixingLengthAux.C:77]:
```suggestion
  if (std::sqrt(min_sq_dist) / _delta <= _von_karman_const_0 / _von_karman_const)
    return _von_karman_const * std::sqrt(min_sq_dist);
  else
    return _von_karman_const_0 * _delta;
```

File: modules/navier_stokes/src/fvkernels/INSFVMixingLengthReynoldsStress.C [hunk 3]
```diff
@@ -100,7 +102,8 @@ INSFVMixingLengthReynoldsStress::computeQpResidual()
               true);
 
   // Compute the eddy diffusivitiy
-  ADReal eddy_diff = velocity_gradient * mixing_len * mixing_len;
+  // ADReal eddy_diff = velocity_gradient * mixing_len * mixing_len;
```
Comment (GiudGiud) [inline modules/navier_stokes/src/fvkernels/INSFVMixingLengthReynoldsStress.C:105]:
```suggestion
```

File: modules/navier_stokes/src/materials/MixingLengthTurbulentViscosityMaterial.C [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+#include "MixingLengthTurbulentViscosityMaterial.h"
+
+registerMooseObject("NavierStokesApp", MixingLengthTurbulentViscosityMaterial);
+
+InputParameters
+MixingLengthTurbulentViscosityMaterial::validParams()
+{
+  InputParameters params = ADMaterial::validParams();
+  params.addClassDescription("Computes the material corresponding to the turbulent viscosity"
+                             "for the mixing length model.");
```
Comment (GiudGiud) [inline modules/navier_stokes/src/materials/MixingLengthTurbulentViscosityMaterial.C:19]: change description
this computes the total viscosity not just the turbulent viscosity

File: modules/navier_stokes/test/tests/finite_volume/ins/wall_distance_capped_mixing_length_aux/tests [hunk 5]
```diff
@@ -0,0 +1,16 @@
+[Tests]
+  issues = '#18666'
+  design = 'navier_stokes/fv.md'
+  [capped_mixing_length]
+    type = 'Exodiff'
+    input = capped_mixing_length.i
+    exodiff = capped_mixing_length_out.e
+    method = "!dbg"
+    requirement = 'The system shall be able to impose turbulent viscosity based on the capped mixing length model.'
```
Comment (GiudGiud) [inline modules/navier_stokes/test/tests/finite_volume/ins/wall_distance_capped_mixing_length_aux/tests:9]: phrasing is weird with impose

File: modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_total_viscosity_material/tests [hunk 6]
```diff
@@ -0,0 +1,16 @@
+[Tests]
+  issues = '#18666'
+  design = 'navier_stokes/fv.md'
+  [capped_mixing_length]
+    type = 'Exodiff'
+    input = mixing_length_total_viscosity.i
+    exodiff = mixing_length_total_viscosity_out.e
+    method = "!dbg"
+    requirement = 'The system shall be able to calculate the total turbulent viscosity based on the capped mixing length model.'
```
Comment (GiudGiud) [inline modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_total_viscosity_material/tests:9]: say it s a material property

File: modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_eddy_viscosity_aux/tests [hunk 7]
```diff
@@ -0,0 +1,16 @@
+[Tests]
+  issues = '#18666'
+  design = 'navier_stokes/fv.md'
+  [capped_mixing_length]
+    type = 'Exodiff'
+    input = mixing_length_eddy_viscosity.i
+    exodiff = mixing_length_eddy_viscosity_out.e
+    method = "!dbg"
+    requirement = 'The system shall be able to compute the turbulent viscosity based on the capped mixing length model.'
```
Comment (GiudGiud) [inline modules/navier_stokes/test/tests/finite_volume/ins/mixing_length_eddy_viscosity_aux/tests:9]: say it s a variable

**Review by GiudGiud — CHANGES_REQUESTED** [review]
ins a FE folder. Use finite_volume/ins for these tests.

**Comment by GiudGiud** [comment]
Any idea as to why your tests are diffing? Are they running fine on your machine?

File: modules/navier_stokes/include/auxkernels/INSFVMixingLengthTurbulentViscosityAux.h [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "AuxKernel.h"
+
+// Computes the value of the eddy viscosity for the mixing length model.
+class INSFVVelocityVariable;
+
+class INSFVMixingLengthTurbulentViscosityAux : public AuxKernel
```
Comment (GiudGiud) [inline modules/navier_stokes/include/auxkernels/INSFVMixingLengthTurbulentViscosityAux.h:17]: this is the format for whole-class header dosctring
```suggestion
class INSFVVelocityVariable;
 
 /*
   * Computes the value of the eddy viscosity for the mixing length model.
   */ 
class INSFVMixingLengthTurbulentViscosityAux : public AuxKernel
```

File: modules/navier_stokes/include/auxkernels/WallDistanceMixingLengthAux.h [hunk 9]
```diff
@@ -10,7 +10,7 @@
 #pragma once
 
 #include "AuxKernel.h"
-
+// Computes the mixing length for the mixing length turbulence model.
```
Comment (GiudGiud) [inline modules/navier_stokes/include/auxkernels/WallDistanceMixingLengthAux.h:13]: same

File: modules/navier_stokes/include/materials/MixingLengthTurbulentViscosityMaterial.h [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ADMaterial.h"
+// Class to compute the total viscosity which considers molecular and
+// mixing length model turbulent viscosity.
```
Comment (GiudGiud) [inline modules/navier_stokes/include/materials/MixingLengthTurbulentViscosityMaterial.h:14]: same

**Review by GiudGiud — APPROVED** [review]

**Comment by GiudGiud** [comment]
@lindsayad this adds some tests that will need some functor material conversion

**Comment by lindsayad** [comment]
:frowning_face:

**Pull request merged**